bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java

```diff
@@ -105,9 +105,19 @@ public void testConstructionNotConnectedExplicitZk() throws Exception {
      * it provides the wrong password or wrong digest.
      */
     @Test
-    public void testBookkeeperPassword() throws Exception {
+    public void testBookkeeperDigestPasswordWithAutoDetection() throws Exception {
+        testBookkeeperDigestPassword(true);
+    }
+
+    @Test
+    public void testBookkeeperDigestPasswordWithoutAutoDetection() throws Exception {
+        testBookkeeperDigestPassword(false);
+    }
+
+    void testBookkeeperDigestPassword(boolean autodetection) throws Exception {
         ClientConfiguration conf = new ClientConfiguration();
         conf.setZkServers(zkUtil.getZooKeeperConnectString());
+        conf.setEnableDigestTypeAutodetection(autodetection);
```

Comment:
I think it is better to enhance MockBookKeeperTestCase and do not export getConf() in the scope of this change.

Reply:
done